test/common.c

```diff
@@ -422,6 +422,16 @@ void replace_strdup(gchar **dst, const gchar *src)
 	*dst = g_strdup(src);
 }
 
+GPtrArray *test_ptr_array_from_strsplit(const gchar *input)
+{
+	GPtrArray *result = g_ptr_array_new_with_free_func(g_free);
+	g_auto(GStrv) strv = g_strsplit(input, ";", 0);
+
+	r_ptr_array_addv(result, strv, FALSE);
```

Comment:
If I'm not mistaken, the elements of `strv` will be freed when leaving the function context and then we return a `GPtrArray` with dangling pointers. Maybe use `TRUE` here?

Reply:
Instead I should only free the array, not the strings.